src/python/zquantum/qcbm/cost_function.py

```diff
@@ -20,22 +59,56 @@ def QCBMCostFunction(
     target_bitstring_distribution: BitstringDistribution,
     gradient_type: str = "finite_difference",
     gradient_kwargs: dict = None,
-):
+) -> CostFunction:
     """Cost function used for evaluating QCBM.
 
     Args:
-        ansatz (zquantum.core.interfaces.ansatz.Ansatz): the ansatz used to construct the variational circuits
-        backend (zquantum.core.interfaces.backend.QuantumBackend): backend used for QCBM evaluation
-        distance_measure (callable): function used to calculate the distance measure
-        distance_measure_parameters (dict): dictionary containing the relevant parameters for the chosen distance measure
-        target_bitstring_distribution (zquantum.core.bitstring_distribution.BitstringDistribution): bistring distribution which QCBM aims to learn
-        gradient_type (str): parameter indicating which type of gradient should be used.
+        ansatz: the ansatz used to construct the variational circuits
+        backend: backend used for QCBM evaluation
+        distance_measure: function used to calculate the distance measure
+        distance_measure_parameters: dictionary containing the relevant parameters for the chosen distance measure
+        target_bitstring_distribution: bistring distribution which QCBM aims to learn
+        gradient_type: parameter indicating which type of gradient should be used.
 
     Returns:
-        Callable that evaluates the parametrized circuit produced by the ansatz with the given parameters and returns
-            the distance between the produced bitstring distribution and the target distribution
+        Callable CostFunction object that evaluates the parametrized circuit produced by the ansatz with the given
+        parameters and returns the distance between the produced bitstring distribution and the target distribution
     """
 
+    DeprecationWarning(
```

Comment:
This looks like a side-effect-less expression, creating an object and that's it. Should we pass it to `warnings.warn`?

Reply:
Fixed in 44bbce7